heltour/tournament/views.py

```diff
@@ -1139,12 +1137,7 @@ def view(self):
     def _common_context(self):
         current_season_list, completed_season_list = _get_season_lists(self.league, active_only=False)
 
-        reg_season = self.season
-        if not self.season.registration_open:
-            for s in self.season.section_list():
-                if s.is_active and s.registration_open:
-                    reg_season = s
-                    break
+        reg_season = Season.get_registration_season(self.league, self.season)
```

Comment:
This logic is wrong here; it may display a link to a completely different season. The old logic should work fine.

Reply:
Yeah, I honestly can't remember even doing this let alone why... Weird.